internal/pkg/agent/application/upgrade/watcher_test.go

```diff
@@ -270,14 +270,25 @@ func TestWatcher_AgentError(t *testing.T) {
 }
 
 func TestWatcher_AgentErrorQuick(t *testing.T) {
-	// test tests for success, which only happens when no error comes in
-	// during this time period
+	// Success only happens when no error comes in during this time period
 	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
 	defer cancel()
 
 	errCh := make(chan error)
-	logger, _ := logger.NewTesting("watcher")
-	w := NewAgentWatcher(errCh, logger, 100*time.Millisecond)
+	log, obs := logger.NewTesting("watcher")
+	defer func() {
+		if t.Failed() {
+			rawLogs := obs.All()
+			for _, rawLog := range rawLogs {
+				msg := fmt.Sprintf("[%s] %s", rawLog.Level, rawLog.Message)
+				for k, v := range rawLog.ContextMap() {
+					msg += fmt.Sprintf("%s=%v", k, v)
+				}
+				t.Log(msg)
+			}
+		}
```

Comment:
I'll add a helper function together with `logger.NewTesting` to pretty print the logs like that on another PR

Reply:
https://github.com/elastic/elastic-agent/pull/5346